api/v1alpha3/awsmachine_webhook.go

```diff
@@ -38,12 +39,21 @@ func (r *AWSMachine) SetupWebhookWithManager(mgr ctrl.Manager) error {
 }
 
 // +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1alpha3-awsmachine,mutating=false,failurePolicy=fail,groups=infrastructure.cluster.x-k8s.io,resources=awsmachines,versions=v1alpha3,name=validation.awsmachine.infrastructure.cluster.x-k8s.io
+// +kubebuilder:webhook:verbs=create;update,path=/mutate-infrastructure-cluster-x-k8s-io-v1alpha3-awsmachine,mutating=true,failurePolicy=fail,groups=infrastructure.cluster.x-k8s.io,resources=awsmachines,versions=v1alpha3,name=default.awsmachine.infrastructure.cluster.x-k8s.io
 
 var _ webhook.Validator = &AWSMachine{}
+var _ webhook.Defaulter = &AWSMachine{}
+
+// Default implements webhook.Defaulter so a mutating webhook will be registered for the type
+func (r *AWSMachine) Default() {
+	if r.Spec.CloudInit.Encrypted == nil {
```

Comment:
Sorry, was thinking specifically about how to handle updates of v1a3 resources, so I had you remove the provider id check. But I was forgetting about v1a2 to v1a3 updates. We need to add something back, but instead of checking provider id, can we check ec2 instance id?